File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.cpp
class MyStack {
private:
    queue<int>q;
    int _top=-1;
    int size=0;
public:
    void push(int x) {
        q.push(x);
        _top=x;
        ++size;
    }
    
    int pop() {
        
        for(int i=0;i<size-2;i++){
            q.push(q.front());
            q.pop();
        }
        if(size>=2){
            _top=q.front();
            q.pop();
            q.push(_top);
        }
        int curr=q.front();
        q.pop();
        
        --size;
        return curr;
    }
    
    int top() {
        return _top;
    }
    
    bool empty() {
        return size==0;
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.java
class MyStack {
    private Queue<Integer>q;
    
    public MyStack() {
        q=new LinkedList<>();
    }
    
    public void push(int x) {
        q.add(x);
    }
    
    public int pop() {
        int size=q.size();
        for(int i=0;i<size-1;++i){
            q.add(q.poll());
        }
        return q.poll();
    }
    
    public int top() {
        int size=q.size();
        for(int i=0;i<size-1;++i){
            q.add(q.poll());
        }
        
        int result=q.poll();
        q.add(result);
        return result;
    }
    
    public boolean empty() {
        return q.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */